torch/lib/TensorLib/Context.cpp

```diff
@@ -0,0 +1,42 @@
+#include "Context.h"
+#include <thread>
+#include <mutex>
+
+#ifdef TENSORLIB_CUDA_ENABLED
+#include "THC/THC.h"
+#include "TensorLib/CUDAGenerator.h"
+#endif
+#include "TensorLib/CPUGenerator.h"
+
+namespace tlib {
+
+Context::Context() {
+
+#ifdef TENSORLIB_CUDA_ENABLED
+  thc_state = THCState_alloc();
```

Comment:
This first commit is mostly to get the changes into CWrap necessary to support both libraries and to test that TensorLib is working, so we haven't implemented the mechanics to get the THCState shared among PyTorch and TensorLib. That will come in a future patch that starts to use it in PyTorch.